src/gsy_e/models/myco_matcher/myco_matcher_forward.py

```diff
@@ -53,7 +53,7 @@ def match_recommendations(self, **kwargs):
         for area_uuid, area_data in self.area_uuid_markets_mapping.items():
             for market_type, matching_algorithm in self._forward_match_algorithms.items():
                 if self._forward_market_counters[market_type].is_time_for_clearing(
-                        area_data["current_time"]):
+                        area_data["current_time"]) and market_type in area_data:
```

Comment:
I am not convinced that this check is needed, since the `_forward_match_algorithms` here and `area_uuid_markets_mapping` generated by [_update_myco_matcher](https://github.com/gridsingularity/gsy-e/blob/39c39daf5727789995c296c60197ae86297df260/src/gsy_e/models/area/__init__.py#L640) should have the same keys.